sihl-queue/src/repo_sql.ml

```diff
@@ -0,0 +1,590 @@
+          status,
+          last_error,
+          last_error_at
+        FROM queue_jobs
+        WHERE
+          status = 'pending'
+          AND next_run_at <= NOW()
+          AND tries < max_tries
+        ORDER BY id DESC
+        |sql}
+  ;;
+
+  let find_workable () =
+    Sihl.Database.query (fun connection ->
+        let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+        Connection.collect_list find_workable_request ()
+        |> Lwt.map Sihl.Database.raise_error)
+  ;;
+
+  let query =
+    Caqti_request.collect
+      Caqti_type.unit
+      job
+      {sql|
+        SELECT
+          uuid,
+          name,
+          input,
+          tries,
+          next_run_at,
+          max_tries,
+          status,
+          last_error,
+          last_error_at
+        FROM queue_jobs
+        ORDER BY next_run_at DESC
+        |sql}
+  ;;
+
+  let query () =
+    Sihl.Database.query' (fun connection ->
+        let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+        Connection.collect_list query ())
+  ;;
+
+  let find_request =
+    Caqti_request.find_opt
+      Caqti_type.string
+      job
+      {sql|
+        SELECT
+          uuid,
+          name,
+          input,
+          tries,
+          next_run_at,
+          max_tries,
+          status,
+          last_error,
+          last_error_at
+        FROM queue_jobs
+        WHERE uuid = ?
+        |sql}
+  ;;
+
+  let find id =
+    Sihl.Database.query' (fun connection ->
+        let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+        Connection.find_opt find_request id)
+  ;;
+
+  let delete_request =
+    Caqti_request.exec
+      Caqti_type.string
+      {sql|
+        DELETE FROM job_queues
+        WHERE uuid = ?
+        |sql}
+  ;;
+
+  let delete (job : Sihl.Contract.Queue.Instance.t) =
+    Sihl.Database.query' (fun connection ->
+        let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+        Connection.exec delete_request job.id)
+  ;;
+
+  let clean_request =
+    Caqti_request.exec Caqti_type.unit "TRUNCATE TABLE queue_jobs;"
+  ;;
+
+  let clean () =
+    Sihl.Database.query (fun connection ->
+        let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+        Connection.exec clean_request () |> Lwt.map Sihl.Database.raise_error)
+  ;;
+
+  module Migration = struct
+    let create_jobs_table =
+      Sihl.Database.Migration.create_step
+        ~label:"create jobs table"
+        {sql|
+         CREATE TABLE IF NOT EXISTS queue_jobs (
+           id serial,
+           uuid uuid NOT NULL,
+           name VARCHAR(128) NOT NULL,
+           input TEXT NULL,
+           tries BIGINT,
+           next_run_at TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP,
+           max_tries BIGINT,
+           status VARCHAR(128) NOT NULL,
+           PRIMARY KEY (id),
+           UNIQUE (uuid)
+         );
+         |sql}
+    ;;
+
+    let set_input_not_null =
```

Comment:
Maybe this here fails too if you didn't set all `NULL` values to `''` prior to running, like in the MariaDB version?

Reply:
@jerben You missed this one. (GitHub hid it and you have to expand manually, pretty annoying)

Reply:
You were right it did fail, I fixed this.